plugins/base/src/main/kotlin/transformers/documentables/SuppressedDocumentableFilterTransformer.kt

```diff
@@ -0,0 +1,51 @@
+package org.jetbrains.dokka.base.transformers.documentables
+
+import org.jetbrains.dokka.DokkaConfiguration
+import org.jetbrains.dokka.DokkaConfiguration.DokkaSourceSet
+import org.jetbrains.dokka.model.*
+import org.jetbrains.dokka.plugability.DokkaContext
+import org.jetbrains.dokka.transformers.documentation.PreMergeDocumentableTransformer
+import java.io.File
+
+class SuppressedDocumentableFilterTransformer(val context: DokkaContext) : PreMergeDocumentableTransformer {
+    override fun invoke(modules: List<DModule>): List<DModule> {
+        return modules.mapNotNull { module -> filterModule(module) }
+    }
+
+    private fun filterModule(module: DModule): DModule? {
+        val packages = module.packages.mapNotNull { pkg -> filterPackage(pkg) }
+        return when {
+            packages == module.packages -> module
+            packages.isEmpty() -> null
+            else -> module.copy(packages = packages)
+        }
+    }
+
+    private fun filterPackage(pkg: DPackage): DPackage? {
+        val options = pkg.perPackageOptions
+        if (options?.suppress == true) {
+            return null
+        }
+
+        val filteredChildren = pkg.children.filter { child -> !isSuppressed(child) }
```

Comment:
I find `filterNot` much more readible.

Reply:
And then you can use `::isSuppressed` reference. But it is a matter of taste, again.

Reply:
Wen't with `pkg.children.filterNot(::isSuppressed)`
I like it. This looks neat 😎

Reply:
✅